drivers/clock_control/nrf_power_clock.c

```diff
@@ -14,38 +14,57 @@
 
 LOG_MODULE_REGISTER(clock_control, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
 
-/* Helper logging macros which prepends device name to the log. */
-#define CLOCK_LOG(lvl, dev, ...) \
-	LOG_##lvl("%s: " GET_ARG1(__VA_ARGS__), dev->config->name \
-			COND_CODE_0(NUM_VA_ARGS_LESS_1(__VA_ARGS__),\
-					(), (, GET_ARGS_LESS_1(__VA_ARGS__))))
-#define ERR(dev, ...) CLOCK_LOG(ERR, dev, __VA_ARGS__)
-#define WRN(dev, ...) CLOCK_LOG(WRN, dev, __VA_ARGS__)
-#define INF(dev, ...) CLOCK_LOG(INF, dev, __VA_ARGS__)
-#define DBG(dev, ...) CLOCK_LOG(DBG, dev, __VA_ARGS__)
+/* Helper logging macros which prepends subsys name to the log. */
+#if IS_ENABLED(CONFIG_LOG)
```

Comment:
This seems a little weird.  Why not `#ifdef CONFIG_LOG` as is usual for preprocessor conditionals?  `IS_ENABLED()` should probably only be used in C expressions.

Reply:
IS_ENABLED is equivalent to `#if defined(CONFIG_LOG) && (CONFIG_LOG == 1)` which is more precise than `#ifdef CONFIG_LOG`. I can switch to `#ifdef` to avoid precedence, that's not a problem. However, i don't see why:
> IS_ENABLED() should probably only be used in C expressions.

Reply:
done

Reply:
> However, i don't see why...

`IS_ENABLED()` does its work with complex macro magic that's necessary only when it's in C language expressions.  For preprocessor use a better check for "defined to a non-zero value" (i.e. explicitly selected as true) is:

    #if (CONFIG_LOG - 0)

but we don't generally see that either because we know that `CONFIG_FOO` values are always either absent or defined to 1, as long as they come from CMake.  They may be defined to an empty token sequence with the intent of making them enable subsequent `#Iifdef ` sections when manually overridden; see e.g. `drivers/media/dvb-frontends/dib0090.c`.

In any case, it's a style issue, and consistency with common practice is generally better as maintainers will be less confused/surprised.  I hadn't seen it used that way before.